server/functions/framework/interpreter_logic.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+
+	"github.com/dolthub/doltgresql/core"
+	"github.com/dolthub/doltgresql/core/id"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// Call runs the contained operations on the given runner.
+func (iFunc InterpretedFunction) Call(ctx *sql.Context, runner analyzer.StatementRunner, paramsAndReturn []*pgtypes.DoltgresType, vals []any) (any, error) {
+	// Set up the initial state of the function
+	statementIdx := -1 // We increment before accessing, so start at -1
+	stack := NewInterpreterStack(runner)
+	// Add the parameters
+	if len(vals) != len(iFunc.ParameterTypes) {
+		return nil, fmt.Errorf("parameter count mismatch: expected `%d` got %d`", len(iFunc.ParameterTypes), len(vals))
+	}
+	for i := range vals {
+		stack.NewVariableWithValue(iFunc.ParameterNames[i], iFunc.ParameterTypes[i], vals[i])
+	}
+	// Run the statements
+	for {
```

Comment:
The cases in this loop should each be their own function / method, would be much easier to scan.

Reply:
I anticipated that you'd probably suggest this, but decided against it for now. It's way easier to debug when everything is on the stack, and I can easily move breakpoints around. I can do it once the interpreter is done for neatness, but going back and forth between a function call and the switch loop is unnecessary complication for now.